src/Platform/Win/WinPlatform.cpp

```diff
@@ -9,66 +9,78 @@
+    const wchar_t *delimiter = wcsstr(path, L"/");
     if (delimiter) {
-        strncpy(current_key, path, delimiter - path);
-        current_key[delimiter - path] = '\0';
+        wcsncpy(current_key, path, delimiter - path);
+        current_key[delimiter - path] = L'\0';
 
-        strcpy(path_tail, delimiter + 1);
+        wcscpy(path_tail, delimiter + 1);
 
         if (!parent_key) {
-            if (iequals(current_key, "HKEY_CLASSES_ROOT"))
+            if (!wcsicmp(current_key, L"HKEY_CLASSES_ROOT"))
                 parent_key = HKEY_CLASSES_ROOT;
-            else if (iequals(current_key, "HKEY_CURRENT_CONFIG"))
+            else if (!wcsicmp(current_key, L"HKEY_CURRENT_CONFIG"))
                 parent_key = HKEY_CURRENT_CONFIG;
-            else if (iequals(current_key, "HKEY_CURRENT_USER"))
+            else if (!wcsicmp(current_key, L"HKEY_CURRENT_USER"))
                 parent_key = HKEY_CURRENT_USER;
-            else if (iequals(current_key, "HKEY_LOCAL_MACHINE"))
+            else if (!wcsicmp(current_key, L"HKEY_LOCAL_MACHINE"))
                 parent_key = HKEY_LOCAL_MACHINE;
-            else if (iequals(current_key, "HKEY_USERS"))
+            else if (!wcsicmp(current_key, L"HKEY_USERS"))
                 parent_key = HKEY_USERS;
             else
                 return false;
 
-            delimiter = strstr(path_tail, "/");
+            delimiter = wcsstr(path_tail, L"/");
             if (delimiter) {
-                strncpy(current_key, path_tail, delimiter - path_tail);
-                current_key[delimiter - path_tail] = '\0';
+                wcsncpy(current_key, path_tail, delimiter - path_tail);
+                current_key[delimiter - path_tail] = L'\0';
 
-                strcpy(path_tail, delimiter + 1);
+                wcscpy(path_tail, delimiter + 1);
             } else {
                 return false;
             }
         }
 
-        if (!strcmp(current_key, "WOW6432Node")) {
+        if (!wcscmp(current_key, L"WOW6432Node")) {
             return OS_GetAppStringRecursive(parent_key, path_tail, out_string, out_string_size, KEY_WOW64_32KEY);
         }
 
         bool result = false;
         HKEY key;
-        if (!RegOpenKeyExA(parent_key, current_key, 0, KEY_READ | flags, &key)) {
+        if (!RegOpenKeyExW(parent_key, current_key, 0, KEY_READ | flags, &key)) {
             result = OS_GetAppStringRecursive(key, path_tail, out_string, out_string_size, 0);
             RegCloseKey(key);
         }
 
         return result;
     } else {
-        DWORD data_size = out_string_size;
-        if (RegQueryValueExA(parent_key, path, nullptr, nullptr, (LPBYTE)out_string, &data_size))
-            return false;
-        return true;
+        wchar_t *tmp = new wchar_t[out_string_size + 1];
+        DWORD data_size = sizeof(wchar_t) * (out_string_size + 1);
+        LSTATUS status = RegQueryValueExW(parent_key, path, nullptr, nullptr, (LPBYTE)tmp, &data_size);
```

Comment:
I've decided to read through the docs for RegQueryValueExW:
> If the data has the REG_SZ, REG_MULTI_SZ or REG_EXPAND_SZ type, the string may not have been stored with the proper terminating null characters. Therefore, even if the function returns ERROR_SUCCESS, the application should ensure that the string is properly terminated before using it; otherwise, it may overwrite a buffer. (Note that REG_MULTI_SZ strings should have two terminating null characters.) One way an application can ensure that the string is properly terminated is to use [RegGetValue](https://learn.microsoft.com/en-us/windows/desktop/api/winreg/nf-winreg-reggetvaluea), which adds terminating null characters if needed.

I suggest we switch over to RegGetValueW and also pass in a type mask (`RRF_RT_REG_SZ | RRF_RT_REG_MULTI_SZ | RRF_RT_REG_EXPAND_SZ`).

Reply:
fair enough, except we don't want MULTI_SZ abomination